cccc/scripts/filter_hf_ds.py

```diff
@@ -0,0 +1,179 @@
+import argparse
+import gzip
+import json
+import logging
+import os
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from functools import partial
+from pathlib import Path
+from urllib.parse import urlparse
+
+from tqdm import tqdm
+
+# Filters the data to only include the URLs in the urls.txt file
+# Requires a local copy of https://huggingface.co/datasets/allenai/dolma-cccc
+# keeps the same folder structure in the destination directory
+
+
+def setup_logging(log_level: str) -> None:
+    """
+    Configures the logging settings.
+    """
+    numeric_level = getattr(logging, log_level.upper(), None)
+    if not isinstance(numeric_level, int):
+        raise ValueError(f"Invalid log level: {log_level}")
+    logging.basicConfig(
+        level=numeric_level,
+        format="%(asctime)s [%(levelname)s] %(processName)s %(message)s",
+        handlers=[logging.StreamHandler()],
+    )
+
+
+def read_file_to_list(file_path):
+    with open(file_path, "r") as file:
+        return [line.strip() for line in file]
+
+
+def process_file(
```

Comment:
These files seem like they should be processable with dolma (they seem to be just be gzipped jsonl files.

We should move this to a dolma processor instead of hand rolling this processing and the parallelism. You should be able to subclass the `licensed_pile.write.ShardParallelProcessor`, override the `process_example` method to call the `filter_condition` function (return the example if it passes and `None` if it fails).

This should simplify things a lot.